docs/proxy/datacenter_proxy/examples.md

```diff
@@ -0,0 +1,668 @@
+            url: "https://www.example.com",
+            proxyUrl: proxyConfiguration.newUrl(),
+        });
+        console.log(body); // returns HTML of returned page
+    } catch (e) {
+        console.error(e);
+    }
+});
+</marked-tab>
+```
+
+### [](#single-ip-address-from-specific-groups) Single IP address from specific groups
+
+The IP address is selected from the `SHADER` and `BUYPROXIES94952` proxy groups.
+
+```marked-tabs
+<marked-tab header="PuppeteerCrawler" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const requestList = await Apify.openRequestList(
+        "my-list", ["http://www.example.com"]
+    );
+    const proxyConfiguration = await Apify.createProxyConfiguration({
+        // if you need to use more then one group
+        // simply add the additional ID to the array below
+        groups: ["SHADER", "BUYPROXIES94952"],
+    });
+
+    const crawler = new Apify.PuppeteerCrawler({
+        requestList,
+        proxyConfiguration,
+        handlePageFunction: async ({ page, request, proxyInfo }) => {
+            return Apify.pushData({ title: await page.title() });
+        },
+        handleFailedRequestFunction: ({ request }) => {
+            console.error("Request failed", request.url, request.errorMessages);
+        },
+    });
+
+    await crawler.run();
+});
+</marked-tab>
+
+
+<marked-tab header="launchPuppeteer()" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const proxyConfiguration = await Apify.createProxyConfiguration({
+        groups: ["SHADER", "BUYPROXIES94952"],
+    });
+
+    const browser = await Apify.launchPuppeteer({
+        proxyUrl: proxyConfiguration.newUrl("my_session"),
+    });
+
+    const page = await browser.newPage();
+
+    await page.goto("http://www.example.com");
+
+    const html = await page.content();
+
+    console.log("HTML:");
+    console.log(html);
+});
+</marked-tab>
+
+
+<marked-tab header="requestAsBrowser()" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const proxyConfiguration = await Apify.createProxyConfiguration({
+        groups: ["SHADER", "BUYPROXIES94952"]
+    });
+    const proxyUrl = proxyConfiguration.newUrl();
+    const url = "https://api.apify.com/v2/browser-info";
+
+    try {
+        const response1 = await Apify.utils.requestAsBrowser({
+            url,
+            proxyUrl,
+            json: true
+        });
+        const response2 = await Apify.utils.requestAsBrowser({
+            url,
+            proxyUrl,
+            json: true
+        });
+        console.log(response1.body.clientIp);
+        console.log("should be different than");
+        console.log(response2.body.clientIp);
+    } catch (e) {
+        console.error(e);
+    }
+});
+</marked-tab>
+```
+
+
+### [](#single-ip-from-a-specific-group-using-puppeteercrawler) Single IP from a specific group using PuppeteerCrawler
```

Comment:
Here it truly is a single IP until failure.

Reply:
In the first example, does each opened browser keep the same IP address until the IP fails?

Reply:
Which example is "the first example"?

Reply:
Oh, wait, I scrolled up to the nearest header, but now I see that there are two different approaches.

In the first example, it will use a random IP address for each new browser. Browsers are typically killed after 100 requests. I would not mention it in the docs, that's an SDK thing. Only mentioning it for context.

In the second example, because of the `maxPoolSize: 1` config, a single IP will be used by all browsers until it fails. Then, all running browsers will get retired and a new IP will be selected, new browsers spawned and they will all use that new IP.